Fixes MYRIAD-114

klucar opened the pull request:
Fixed gulp build to not include local paths and gradle-ized build.
Removed public/ from git repo and added it to .gitignore.

smarella commented on myriad-scheduler/build.gradle:
```diff
@@ -69,3 +74,36 @@ sourceSets {
+    workDir = file("${project.projectDir}/src/main/resources/webapp")
+
+    // Whether colors should output on the terminal
+    colors = false
```
No colors?

klucar replied:
ha! sometimes colors just print weird control characters if your terminal can't handle it. you're not missing anything too exciting.

smarella commented:
Tested it on my dev setup, but the RM can't come up. It throws me the following null pointer:

```
Caused by: java.lang.NullPointerException
        at org.apache.myriad.webapp.MyriadWebServer.start(MyriadWebServer.java:64)
        at org.apache.myriad.Main.initWebApp(Main.java:119)
        at org.apache.myriad.Main.run(Main.java:91)
        at org.apache.myriad.Main.initialize(Main.java:76)
        at org.apache.myriad.scheduler.yarn.interceptor.MyriadInitializationInterceptor.init(MyriadInitializationInterceptor.java:51)
        ... 16 more
```
The corresponding code in the webserver is:
```java
    String staticDir = this.getClass().getClassLoader().getResource("webapp/public").toExternalForm();
```
The myriad-scheduler-0.1.jar doesn't have "webapp/public". The build doesn't seem to copy "webapp/public" under the build directory.

klucar commented:
I saw that once but couldn't reproduce it. I figured it was just from me
trying to get everything working. A few questions: did you gradle or
gradlew? Were the files generated in src/main/resources/webapp/public and
just didn't make it into the jar? I'll try to reproduce again this morning.

On Thursday, October 29, 2015, Santosh Marella <notifications@github.com>
wrote:

> Tested it on my dev setup, but the RM can't come up. It throws me the
> following null pointer:
>
> Caused by: java.lang.NullPointerException
>         at org.apache.myriad.webapp.MyriadWebServer.start(MyriadWebServer.java:64)
>         at org.apache.myriad.Main.initWebApp(Main.java:119)
>         at org.apache.myriad.Main.run(Main.java:91)
>         at org.apache.myriad.Main.initialize(Main.java:76)
>         at org.apache.myriad.scheduler.yarn.interceptor.MyriadInitializationInterceptor.init(MyriadInitializationInterceptor.java:51)
>         ... 16 more
>
> The corresponding code in the webserver is:
>
>     String staticDir = this.getClass().getClassLoader().getResource("webapp/public").toExternalForm();
>
> The myriad-scheduler-0.1.jar doesn't have "webapp/public". The build
> doesn't seem to copy "webapp/public" under the build directory.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-myriad/pull/28#issuecomment-152357160>
> .
>

smarella commented:
I did "gradlew".

Yes. The files were generated under src/main/resources/webapp/public, but
didn't get copied over to myriad-scheduler/build/resources/main/webapp and
subsequently didn't make it into the myriad-scheduler-0.1.jar.

Santosh

On Fri, Oct 30, 2015 at 5:04 AM, Jim Klucar <notifications@github.com>
wrote:

> I saw that once but couldn't reproduce it. I figured it was just from me
> trying to get everything working. A few questions: did you gradle or
> gradlew? Were the files generated in src/main/resources/webapp/public and
> just didn't make it into the jar? I'll try to reproduce again this morning.
>
> On Thursday, October 29, 2015, Santosh Marella <notifications@github.com>
> wrote:
>
> > Tested it on my dev setup, but the RM can't come up. It throws me the
> > following null pointer:
> >
> > Caused by: java.lang.NullPointerException
> > at
> org.apache.myriad.webapp.MyriadWebServer.start(MyriadWebServer.java:64)
> > at org.apache.myriad.Main.initWebApp(Main.java:119)
> > at org.apache.myriad.Main.run(Main.java:91)
> > at org.apache.myriad.Main.initialize(Main.java:76)
> > at
> org.apache.myriad.scheduler.yarn.interceptor.MyriadInitializationInterceptor.init(MyriadInitializationInterceptor.java:51)
> > ... 16 more
> >
> > The corresponding code in the webserver is:
> >
> > String staticDir =
> this.getClass().getClassLoader().getResource("webapp/public").toExternalForm();
> >
> > The myriad-scheduler-0.1.jar doesn't have "webapp/public". The build
> > doesn't seem to copy "webapp/public" under the build directory.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/apache/incubator-myriad/pull/28#issuecomment-152357160>
> > .
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-myriad/pull/28#issuecomment-152506635>
> .
>

klucar commented:
I updated the PR. I moved the web app build to before process resources.
Brandon and I both tried it and saw the files in the jar now.

On Friday, October 30, 2015, Santosh Marella <notifications@github.com>
wrote:

> I did "gradlew".
>
> Yes. The files were generated under src/main/resources/webapp/public, but
> didn't get copied over to myriad-scheduler/build/resources/main/webapp and
> subsequently didn't make it into the myriad-scheduler-0.1.jar.
>
> Santosh
>
> On Fri, Oct 30, 2015 at 5:04 AM, Jim Klucar <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>>
> wrote:
>
> > I saw that once but couldn't reproduce it. I figured it was just from me
> > trying to get everything working. A few questions: did you gradle or
> > gradlew? Were the files generated in src/main/resources/webapp/public and
> > just didn't make it into the jar? I'll try to reproduce again this
> morning.
> >
> > On Thursday, October 29, 2015, Santosh Marella <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>>
> > wrote:
> >
> > > Tested it on my dev setup, but the RM can't come up. It throws me the
> > > following null pointer:
> > >
> > > Caused by: java.lang.NullPointerException
> > > at
> > org.apache.myriad.webapp.MyriadWebServer.start(MyriadWebServer.java:64)
> > > at org.apache.myriad.Main.initWebApp(Main.java:119)
> > > at org.apache.myriad.Main.run(Main.java:91)
> > > at org.apache.myriad.Main.initialize(Main.java:76)
> > > at
> >
> org.apache.myriad.scheduler.yarn.interceptor.MyriadInitializationInterceptor.init(MyriadInitializationInterceptor.java:51)
> > > ... 16 more
> > >
> > > The corresponding code in the webserver is:
> > >
> > > String staticDir =
> >
> this.getClass().getClassLoader().getResource("webapp/public").toExternalForm();
> > >
> > > The myriad-scheduler-0.1.jar doesn't have "webapp/public". The build
> > > doesn't seem to copy "webapp/public" under the build directory.
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
> https://github.com/apache/incubator-myriad/pull/28#issuecomment-152357160>
> > > .
> >
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/apache/incubator-myriad/pull/28#issuecomment-152506635>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-myriad/pull/28#issuecomment-152567043>
> .
>

smarella commented:
awesome. I'll test it out again before merging.

Santosh

On Fri, Oct 30, 2015 at 12:53 PM, Jim Klucar <notifications@github.com>
wrote:

> I updated the PR. I moved the web app build to before process resources.
> Brandon and I both tried it and saw the files in the jar now.
>
> On Friday, October 30, 2015, Santosh Marella <notifications@github.com>
> wrote:
>
> > I did "gradlew".
> >
> > Yes. The files were generated under src/main/resources/webapp/public, but
> > didn't get copied over to myriad-scheduler/build/resources/main/webapp
> and
> > subsequently didn't make it into the myriad-scheduler-0.1.jar.
> >
> > Santosh
> >
> > On Fri, Oct 30, 2015 at 5:04 AM, Jim Klucar <notifications@github.com
> > <javascript:_e(%7B%7D,'cvml','notifications@github.com');>>
> > wrote:
> >
> > > I saw that once but couldn't reproduce it. I figured it was just from
> me
> > > trying to get everything working. A few questions: did you gradle or
> > > gradlew? Were the files generated in src/main/resources/webapp/public
> and
> > > just didn't make it into the jar? I'll try to reproduce again this
> > morning.
> > >
> > > On Thursday, October 29, 2015, Santosh Marella <
> notifications@github.com
> > <javascript:_e(%7B%7D,'cvml','notifications@github.com');>>
> > > wrote:
> > >
> > > > Tested it on my dev setup, but the RM can't come up. It throws me the
> > > > following null pointer:
> > > >
> > > > Caused by: java.lang.NullPointerException
> > > > at
> > > org.apache.myriad.webapp.MyriadWebServer.start(MyriadWebServer.java:64)
> > > > at org.apache.myriad.Main.initWebApp(Main.java:119)
> > > > at org.apache.myriad.Main.run(Main.java:91)
> > > > at org.apache.myriad.Main.initialize(Main.java:76)
> > > > at
> > >
> >
> org.apache.myriad.scheduler.yarn.interceptor.MyriadInitializationInterceptor.init(MyriadInitializationInterceptor.java:51)
> > > > ... 16 more
> > > >
> > > > The corresponding code in the webserver is:
> > > >
> > > > String staticDir =
> > >
> >
> this.getClass().getClassLoader().getResource("webapp/public").toExternalForm();
> > > >
> > > > The myriad-scheduler-0.1.jar doesn't have "webapp/public". The build
> > > > doesn't seem to copy "webapp/public" under the build directory.
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > >
> >
> https://github.com/apache/incubator-myriad/pull/28#issuecomment-152357160>
> > > > .
> > >
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
> https://github.com/apache/incubator-myriad/pull/28#issuecomment-152506635>
> > > .
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/apache/incubator-myriad/pull/28#issuecomment-152567043>
> > .
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-myriad/pull/28#issuecomment-152636073>
> .
>

smarella commented:
Tested it out. The build is good, RM comes up alright. But the UI doesn't seem to render. Only the favicon and the page title seem to appear.

klucar commented:
I found an error in the shutdown component that kept it from rendering. I actually think I want to re-do the interface to make it more like the website design, but that can wait a bit.

smarella commented:
@klucar - any chance this could be fixed today? This is a blocker for 0.1.0.

klucar commented:
I thought I fixed it, maybe I didn't push. It was a simple error in the
JavaScript. Check the PR. I'm on mobile now but can look at it later.

On Tuesday, November 3, 2015, Santosh Marella <notifications@github.com>
wrote:

> @klucar <https://github.com/klucar> - any chance this could be fixed
> today? This is a blocker for 0.1.0.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-myriad/pull/28#issuecomment-153410420>
> .
>

smarella commented:
@klucar - you are right. I see a commit added to the PR. Not sure why I don't receive the git notifications, though. I'll test this out locally and merge in if the UI renders correctly. Thanks!
